erpnext/projects/doctype/timesheet/timesheet.js

```diff
@@ -101,12 +101,60 @@ frappe.ui.form.on("Timesheet", {
 			});
 	},
 
-	make_invoice: function(frm) {
+	make_invoice: async function(frm) {
+		let customers = [];
+		const billable_projects = frm.doc.time_logs
+			.filter(log => log.billable)
+			.map(log => log.project)
+			.filter(Boolean);
+
+		// find unique customers to bill from linked projects
+		if (billable_projects.length > 0) {
+			for (let project of billable_projects) {
+				await frappe.db.get_value("Project", { "name": project }, "customer")
+					.then((r) => {
+						customers.push(r.message.customer);
+					})
+			}
+		}
+
+		// set default customer and project if single value found
+		let project;
+		let customer;
+
+		if (customers.length == 1) {
+			customer = customers[0];
```

Comment:
what's the behavior if more than one customer is found? is that even a posibility?

Reply:
My first idea was that we could set the query for each field with the options, but I didn't want to restrict the user from selecting particular customers and projects, and select an option in the dialog for them